### PR TITLE
convert to legacy format to support Nx 11

### PR DESCRIPTION
**Pull request by sandikbarr**
converts workspace, package.json, generators, and executors to Legacy formats to support Nx11

### CODE REVIEW

File: libs/server/src/lib/utils/utils.ts [hunk 1]
```diff
@@ -290,3 +290,38 @@ export function getPrimitiveValue(value: any): string | undefined {
     return undefined;
   }
 }
+
+function renameProperty(obj: any, from: string, to: string) {
```
Comment (vsavkin) [inline libs/server/src/lib/utils/utils.ts:294]: We do it in this way in the Nx repo itself cause we need to preserve the order of properties (cause we are serializing the json back to the file). You don't do the serialization, so the order should not matter. So you can do the rename as follows:

```
obj[to] = obj[from]
delete obj[from]
```

File: libs/server/src/lib/utils/read-projects.ts [hunk 2]
```diff
@@ -74,14 +75,14 @@ export async function readBuilderSchema(
     path.join(npmPackage, 'package.json'),
     path.join(basedir, 'node_modules')
   );
-  const b = packageJson.json.builders;
+  const b = toLegacyFormat(packageJson.json).builders;
```
Comment (vsavkin) [inline libs/server/src/lib/utils/read-projects.ts:78]: toLegacyFormat only updates angular.json/workspace.json. You probably either have a similar one to package.json that renames generators into schematics and executors into builders or do it on the spot here.

File: libs/server/src/lib/utils/read-projects.ts [hunk 3]
```diff
@@ -74,14 +75,14 @@ export async function readBuilderSchema(
...
   const buildersPath = b.startsWith('.') ? b : `./${b}`;
   const buildersJson = readAndCacheJsonFile(
     buildersPath,
     path.dirname(packageJson.path)
   );
 
-  const builderDef = buildersJson.json.builders[builderName];
+  const builderDef = toLegacyFormat(buildersJson.json).builders[builderName];
```
Comment (vsavkin) [inline libs/server/src/lib/utils/read-projects.ts:85]: See the comment above. toLegacyFormat only updates workspace.json/angular.json. You need to a different toLegacyFormat here.

**Review by vsavkin — CHANGES_REQUESTED** [review]
You need to have 4 normalization functions instead of one. You have one for workspace.json, but you also need one for package.json, generators.json and executors.json. The missing three should be very small (3 lines of code each). So you don't really have to have them as functions, you can also inline them.

File: libs/server/src/lib/utils/utils.ts [hunk 4]
```diff
@@ -290,3 +290,33 @@ export function getPrimitiveValue(value: any): string | undefined {
...
+  Object.keys(obj).forEach(k => {
```
Comment (vsavkin) [inline libs/server/src/lib/utils/utils.ts:295]: you don't need to loop, right? you aren't using `k` for anything.

Comment (sandikbarr) [reply libs/server/src/lib/utils/utils.ts:295]: gah, sorry yeah. will fix.

**Pull request merged**